src/core/macro.scala

```diff
@@ -12,6 +12,12 @@ object Macro:
   inline def paramTypeAnns[T]: List[(String, List[Any])] = ${paramTypeAnns[T]}
   inline def repeated[T]: List[(String, Boolean)] = ${repeated[T]}
   inline def typeInfo[T]: TypeInfo = ${typeInfo[T]}
+  inline def summonOption[T]: Option[T] = ${summonOptionImpl[T]}
+
+  def summonOptionImpl[T: Type](using Quotes): Expr[Option[T]] =
+    Expr.summon[T] match
```

Comment:
can't this simply be `Expr.summon[T]` without the match? :)

Reply:
The generics are switched. `Expr[Option[T]]` vs `Option[Expr[T]]`

Reply:
ah 🤦